src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -3480,5 +3486,852 @@ void F()
+        a = [A, B] (x, y) => { };
+        a = ([A] x, [B] y) => { };
+        a = (object x, [A][B] object y) => { };
+    }
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (9,13): error CS8652: The feature 'lambda attributes' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         a = [A, B] (x, y) => { };
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "[A, B]").WithArguments("lambda attributes").WithLocation(9, 13),
+                // (10,14): error CS8652: The feature 'lambda attributes' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         a = ([A] x, [B] y) => { };
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "[A]").WithArguments("lambda attributes").WithLocation(10, 14),
+                // (10,21): error CS8652: The feature 'lambda attributes' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         a = ([A] x, [B] y) => { };
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "[B]").WithArguments("lambda attributes").WithLocation(10, 21),
+                // (11,24): error CS8652: The feature 'lambda attributes' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         a = (object x, [A][B] object y) => { };
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "[A]").WithArguments("lambda attributes").WithLocation(11, 24),
+                // (11,27): error CS8652: The feature 'lambda attributes' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         a = (object x, [A][B] object y) => { };
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "[B]").WithArguments("lambda attributes").WithLocation(11, 27));
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void LambdaAttributes_03()
+        {
+            var source =
+@"using System;
+class AAttribute : Attribute { }
+class BAttribute : Attribute { }
+class C
+{
+    static void Main()
+    {
+        Action<object, object> a = delegate (object x, [A][B] object y) { };
+        Func<object, object> f = [A][B] x => x;
+    }
+}";
+
+            var expectedDiagnostics = new[]
+            {
+                // (8,56): error CS8916: Attributes require a lambda expression with a parenthesized parameter list.
+                //         Action<object, object> a = delegate (object x, [A][B] object y) { };
+                Diagnostic(ErrorCode.ERR_AttributesRequireParenthesizedLambdaExpression, "[A]").WithLocation(8, 56),
```

Comment:
Here the message doesn't make sense, as the parameter list is already parenthesized :-/

Reply:
Yes, the parameter list is parenthesized, but the message also indicates the expression should be a lambda expression.

Reply:
I've reverted the error message for anonymous methods.